packages/perspective-viewer-d3fc/src/js/legend/scrollableLegend.js

```diff
@@ -0,0 +1,92 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2017, the Perspective Authors.
+ *
+ * This file is part of the Perspective library, distributed under the terms of
+ * the Apache License 2.0.  The full license can be found in the LICENSE file.
+ *
+ */
+import * as d3Legend from "d3-svg-legend";
+import {rebindAll} from "d3fc";
+import {areArraysEqualSimple, getOrCreateElement} from "../utils/utils";
+import legendControlsTemplate from "../../html/legend-controls.html";
+
+export default () => {
+    const legend = d3Legend.legendColor();
+    let domain = [];
+    let pageCount = 1;
+    let pageSize = 15;
+    let pageIndex = 0;
+
+    const scrollableLegend = selection => {
+        const newDomain = legend.scale().domain();
+        if (!areArraysEqualSimple(domain, newDomain)) {
+            pageIndex = 0;
+            domain = newDomain;
+        }
+        pageCount = Math.ceil(domain.length / pageSize);
+
+        render(selection);
+    };
+
+    const render = selection => {
+        renderControls(selection);
+        renderLegend(selection);
+    };
+
+    const renderControls = selection => {
+        const controls = getLegendControls(selection);
+        controls.style("display", pageCount <= 1 ? "none" : "block");

```

Comment:
Do we need to continue from this point if `pageCount <= 1`?

Reply:
We could exit here, but it won't change the behaviour. I think I'd prefer to leave it as it is, so the controls are actually "correct" even though they're not visible.